Sources/PenguinStructures/NumberOperations.swift

```diff
@@ -0,0 +1,120 @@
+// Copyright 2020 Penguin Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extension BinaryInteger {
+  /// Returns a collection of the positive integers that are coprime with `self`.
+  ///
+  /// Definition: Two numbers are coprime if their GCD is 1.
+  public var positiveCoprimes: PositiveCoprimes<Self> { .init(self) }
+
+  /// Returns positive integers that are coprime with, and smaller than, `self`.
+  ///
+  /// - SeeAlso: `positiveCoprimes`.
+  public var smallerPositiveCoprimes: [Self] {
+    var positiveSelf = self
+    if positiveSelf < 0 {
+      positiveSelf *= -1  // Workaround for lack of `Swift.abs` on `BinaryInteger`.
+    }
```

Comment:
This is a weird definition of "smaller" that apparently(?) from reading the code implies absolute value.
Why not just have the array be empty when self is <= 0?
Why not return PositiveCoprimes.SubSequence?
How can this possibly even compile?  `.prefix(…)` is supposed to return a SubSequence, i.e. slice.

Reply:
> Why not just have the array be empty when self is <= 0?

Yeah, that's reasonable. Fixed.

> How can this possibly even compile? `.prefix(…)` is supposed to return a SubSequence, i.e. slice.

As it turns out, IIUC it's calling [`Sequence.prefix(while:)`](https://developer.apple.com/documentation/swift/sequence/3128810-prefix)

As for why not return `PositiveCoprimes.SubSequence`, the concern I have is that this will cause us to traverse the collection at least twice: once to find the `endIndex`, and then a second time to access the elements. Of course, we could make this be a `LazyPrefixWhileSequence<SubSequence>`, but this forces closure allocation. 

The final option would be to do something quite fancy with the `SubSequence` type. My preference is to avoid getting that fancy without more experience to understand what operations we'd like to support efficiently (as we only get one `SubSequence` type).